File: it/default_naming/structures/default/python/upper_snake.py
class Value:
  def __init__(self, _foo_bar):
    self._foo_bar = _foo_bar

  @property
  def foo_bar(self):
    return self._foo_bar

  @staticmethod
  def decode(data):
    f_foo_bar = data["FOO_BAR"]

    if not isinstance(f_foo_bar, unicode):
      raise Exception("not a string")

    return Value(f_foo_bar)

  def encode(self):
    data = dict()

    if self._foo_bar is None:
      raise Exception("FOO_BAR: is a required field")

    data["FOO_BAR"] = self._foo_bar

    return data

  def __repr__(self):
    return "<Value foo_bar:{!r}>".format(self._foo_bar)

File: it/default_naming/structures/default/dart/upper_snake.dart
class Value {
  String fooBar;

  Value(
    this.fooBar
  );

  static Value decode(dynamic _data_dyn) {
    if (!(_data_dyn is Map<String, dynamic>)) {
      throw 'expected Map<String, dynamic>, but got: $_data_dyn';
    }

    Map<String, dynamic> _data = _data_dyn;

    var fooBar_dyn = _data["FOO_BAR"];
    if (fooBar_dyn == null) {
      throw "expected value but was null";
    }
    if (!(fooBar_dyn is String)) {
      throw 'expected String, but was: $fooBar_dyn';
    }
    final String fooBar = fooBar_dyn;

    return Value(fooBar);
  }

  Map<String, dynamic> encode() {
    Map<String, dynamic> _data = Map();

    _data["FOO_BAR"] = this.fooBar;

    return _data;
  }
}
